packages/vue-instantsearch/src/util/vue-compat/index-vue3.js

```diff
@@ -9,13 +9,34 @@ export { Vue, Vue2, isVue2, isVue3 };
 
 export function renderCompat(fn) {
   function h(tag, props, children) {
-    if (typeof props === 'object' && (props.attrs || props.props)) {
+    if (
+      typeof props === 'object' &&
+      (props.attrs || props.props || props.scopedSlots || props.on)
+    ) {
       // In vue 3, we no longer wrap with `attrs` or `props` key.
-      const flatProps = Object.assign({}, props, props.attrs, props.props);
+      const flatProps = Object.assign(
+        {},
+        props,
+        props.attrs,
+        props.props,
+        Object.keys(props.on || {}).reduce((acc, key) => {
+          // eslint-disable-next-line no-param-reassign
+          acc[`on${key[0].toUpperCase()}${key.slice(1)}`] = props.on[key];
+          return acc;
+        }, {})
+      );
       delete flatProps.attrs;
       delete flatProps.props;
+      delete flatProps.scopedSlots;
+      delete flatProps.on;
 
-      return Vue.h(tag, flatProps, children);
+      return Vue.h(
+        tag,
+        flatProps,
+        props.scopedSlots
+          ? { default: () => children, ...props.scopedSlots }
```

Comment:
This needs to be in a transpilable form, Vue InstantSearch doesn’t have object spread.